src/dbnode/integration/dynamic_namespace_add_test.go

```diff
@@ -86,11 +91,40 @@ func TestDynamicNamespaceAdd(t *testing.T) {
 	require.NoError(t, testSetup.startServer())
 
 	// Stop the server
+	stopped := false
 	defer func() {
+		stopped = true
 		require.NoError(t, testSetup.stopServer())
 		log.Info("server is now down")
 	}()
 
+	// NB(bodu): concurrently do work on the leaseManager to ensure no race cond deadlock regressions by
+	// calling OpenLease or OpenLatestLease (blocked on DB lock).
+	var wg sync.WaitGroup
+	leaser := block.NewMockLeaser(ctrl)
+	leaseState := block.LeaseState{}
+	for i := 0; i < 100; i++ {
+		leaseDescriptor := block.LeaseDescriptor{
+			Namespace:  ns0.ID(),
+			Shard:      uint32(0),
+			BlockStart: time.Now().Truncate(ns0.Options().RetentionOptions().BlockSize()),
+		}
+		wg.Add(2)
+		go func() {
+			wg.Done()
```

Comment:
Should these be `Done` after the OpenLease call?

Reply:
We just want to wait for the goroutine to bootstrap. The calls are done in a loop till the server is called. They were the original source of the deadlock.